src/features/channels/channelsSlice.ts

```diff
@@ -24,13 +24,20 @@ export const channelsSlice = createSlice({
         {
           payload: {
             data: {
-              specification: {
-                spec: { channels }
-              }
+              specification: { spec }
             }
           }
         }
       ) => {
+        // channels can be undefined if a lockfile specification is provided,
+        // try getting channels from metadata in that case
+        const channels =
+          spec?.channels ??
+          spec?.lockfile?.metadata?.channels?.map(
+            (channel: any) => channel?.url
+          ) ??
+          [];
```

Comment:
This and similar code for deps is what populates GUI channels and dependencies lists on env save.